File: FangZhiHu/CommonLib/src/main/java/qj/ccmtjf/com/lib/api/ApiResponse.java
package qj.ccmtjf.com.lib.api;

/**
 * Common class used by API responses.
 *
 * @param <T> 范型
 *            Created by liyanxi on 10/26/17.
 *            Copyright (c) 2017 www.zhengshijr.com. All rights reserved.
 */

public class ApiResponse<T> {


    private int code;    // 返回码，0为成功
    private String message;      // 返回信息
    private T data;           // 单个对象/数组对象
    public int errorCode;//: 0,
    public String errorMsg;//: ""

    public ApiResponse(int status, String message) {
        this.code = status;
        this.message = message;
    }

    public ApiResponse(T data, String message, int status) {
        this.data = data;
        this.message = message;
        this.code = status;
    }

    public ApiResponse(String message) {
        this.message = message;
    }

    public boolean isSuccess() {
        return code == 0;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "ApiResponse{" +
                "code=" + code +
                ", message='" + message + '\'' +
                ", data=" + data +
                ", errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                '}';
    }
}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/entity/ApiResponse.kt
package network.entity

/**
 * Created by tongqinyuan on 2018/3/5.
 */
class ApiResponse<T> {


    var data: T? = null
    var code: Int = 0    // 返回码，0为成功
    var message: String? = null
    var status: Int = 0
    var errorCode: Int = 0
    var errorMsg: String? = null
    // 返回信息
    // 单个对象/数组对象

    constructor(data: T, message: String, status: Int) {
        this.data = data
        this.message = message
        this.code = status
    }

    constructor(status: Int, message: String) {
        this.code = status
        this.message = message
    }

    constructor(message: String) {
        this.message = message
    }

    override fun toString(): String {
        return "ApiResponse(data=$data, code=$code, message=$message, status=$status, errorCode=$errorCode, errorMsg=$errorMsg)"
    }


}